src/components/general/character/Poses.jsx

```diff
@@ -1,56 +1,61 @@
-import React, {useState, useRef} from 'react';
+import React, { useState, useRef } from 'react';
 import classnames from 'classnames';
 
-import {setFacePoseValue} from './Emotions';
+import { setFacePoseValue } from './Emotions';
 import styles from './poses.module.css';
 
 import metaversefile from 'metaversefile';
-import {emoteAnimations} from '../../../../avatars/animationHelpers.js';
+import { emoteAnimations } from '../../../../avatars/animationHelpers.js';
 import emotes from './emotes.json';
+import fallbackEmotes from "./fallback_emotes.json";
 
-//
+//The trigger emote function is modified to trigger for the required player.
 
 let emoteTimeout = null;
-export const triggerEmote = emoteName => {
-  const emoteHardName = emoteName.replace(/Soft$/, '');
-  const emote = emotes.find(emote => emote.name === emoteHardName);
-  const {emotion} = emote;
-  
-  // clear old emote
-  const localPlayer = metaversefile.useLocalPlayer();
-  localPlayer.removeAction('emote');
-  if (emoteTimeout) {
-    clearTimeout(emoteTimeout);
-    emoteTimeout = null;
-  }
+export const triggerEmote = (emoteName, player = null) => {
+    const emoteHardName = emoteName.replace(/Soft$/, '');
+    const emoteFinalName = fallbackEmotes[emoteHardName] || emoteHardName;
+    const emote = emotes.find(emote => emote.name === emoteFinalName);
+    if (emote === undefined){
```

Comment:
This should never be hit IMO. The emote passed in here should be valid. It should be the caller's responsibility to filter it.